### ISSUE TITLE
setVelocitiesToTemperature() sets nan velocities if positions are not set

### ISSUE THREAD

**Issue by andrrizzi** [pandegroup/openmm#1687]
This script
```python
from simtk import openmm, unit
from openmmtools import testsystems
toluene_test = testsystems.TolueneVacuum()
integrator = openmm.VerletIntegrator(1.0*unit.femtosecond)

context = openmm.Context(toluene_test.system, integrator)
print(context.getState(getVelocities=True).getVelocities()[0])

context.setVelocitiesToTemperature(300*unit.kelvin)
print(context.getState(getVelocities=True).getVelocities()[0])

context.setPositions(toluene_test.positions)
print(context.getState(getVelocities=True).getVelocities()[0])

context.setVelocitiesToTemperature(300*unit.kelvin)
print(context.getState(getVelocities=True).getVelocities()[0])
```
outputs
```
(0.0, 0.0, 0.0) nm/ps
(nan, nan, nan) nm/ps
(nan, nan, nan) nm/ps
(0.1276603788137436, -0.2770220637321472, -0.8963528275489807) nm/ps
```
Is this expected behavior? Is there a fundamental reason why velocities cannot be set before positions? Thanks!

**Comment by peastman** [comment]
If your system contains any constraints, it needs to apply those constraints to the velocities.  The net velocity along any constrained direction needs to be 0, or else you can run into various problems.  But if the positions haven't been set yet, the constrained directions are undefined leading to this error.

**Comment by andrrizzi** [comment]
l see, thank you for the explanation!

**Comment by jchodera** [comment]
Shouldn't this raise an `Exception` rather than set things to `nan`?

**Comment by peastman** [comment]
That would make sense.  I'll reopen this.

**Comment by peastman** [comment]
This was fixed by #1711.
